### ISSUE TITLE
View logs in realtime

### ISSUE THREAD

**Issue by kraenhansen** [github/vscode-github-actions#126]
**Is your feature request related to a problem? Please describe.**
While it's currently possible to view a log after a job has completed, I would love to be able to follow the logs of an action directly from VSCode in realtime.

**Describe the solution you'd like**
Double-clicking a running job / step, I would like the editor (or an output tab) to show the log and see new entries being streamed in.

**Additional context**
None.

**Comment by felipesu19** [comment]
Unfortunately this is not a feature we are ready to commit to or to put on our road map at this time. We appreciate the suggestion, and will re-open if that changes!

**Comment by kraenhansen** [comment]
@felipesu19 I appreciate the honesty. Is it mostly the implementation and would you be interested in merging a PR for this or is it also the eventual maintenance that you're not interested in?

**Comment by felipesu19** [comment]
> @felipesu19 I appreciate the honesty. Is it mostly the implementation and would you be interested in merging a PR for this or is it also the eventual maintenance or feature as such that you're not interested in?

It's a complicated discussion, we *definitely* don't have the bandwidth to implement it ourselves. The maintenance/usability questions are tricky. Essentially we'd have to answer: 

1. Does this increase the scope of the extension, e.g. do we need more permissions, or system access in ways we didn't before? (I think we would need more actions permissions for this one, but unsure), if we do, is the tradeoff worth it for customers? For this feature i'd probably argue "no", but I don't have a super strong stance on it, its my initial thoughts. 

2. What's the maintenance burden? I don't know if the actions logs api is going to change/what its limitations are/if we'd need to poll it, but changes to that would require updating the feature. It'd also require knowing how people are going to use it and under what circumstances it would break, e.g. will it cause someone's corporate proxy to freak out if we have to poll an endpoint every x amount of time? 

3. Do we need to add settings to turn it on and off? 

I'm sure there's a few questions i haven't considered in here, and I absolutely don't want to discourage contribution, so if you want to consider opening a PR for it, absolutely go for it, but I would suggest a robust PR description, or even better a new issue where you lay out "here's the PR, and here are the considerations around it that don't make it too much of a stretch/prove that its maintainable". We're very committed to this extension being a first class experience, and so we have to be a little selective on what we're doing. One of those tradeoffs is "what does a feature mean for the medium-long term". 

I think this feature and the notification feature are both "doable" for sure, they just need to be properly thought through.

**Comment by kraenhansen** [comment]
@felipesu19 thanks for the clarification. I'll get back if/when this grows on me, but only after doing a bit more digging in the underlying API and the current state of the extension 👍
